src/StochasticStyles/spawning.jl

```diff
@@ -199,11 +199,11 @@ struct SingleSpawn{T} <: SpawningStrategy
     SingleSpawn(threshold::T=0.0) where {T} = new{T}(threshold)
 end
 
-@inline function spawn!(s::SingleSpawn, w, offdiags::AbstractVector, add, val, boost=1)
+@inline function spawn!(s::SingleSpawn, w, column::AbstractVector, add, val, boost=1)
     if iszero(val)
         return (1, zero(valtype(w)))
     else
-        new_add, prob, mat_elem = random_offdiagonal(offdiags)
+        new_add, prob, mat_elem = random_element(column)
```

Comment:
Isn't it wrong to replace `random_offdiagonal` by `random_element` here? The algorithm previously assumed that the diagonal element was taken care of separately.